[Version Update] Fix plugin for Gnome 49 Compatibility

aoiwelle opened the pull request:
Unfortunately, the plugin used library versions that changed in Gnome 49, so this migrates to use the new methods.

* [`Clutter.ClickAction()` -> `Clutter.ClickGesture()` and `Clutter.LongPressGesture()`](https://gjs.guide/extensions/upgrading/gnome-shell-49.html#clutter)
* [`OsdWindowManager.show()` -> `OsdWindowManager.showAll()`](https://gjs.guide/extensions/upgrading/gnome-shell-49.html#osdwindowmanager)
* bump version number in metadata.json
This is intended to fix [Issue #43 ](https://github.com/joaophi/tailscale-gnome-qs/issues/43)

aoiwelle committed:
[Version Update] Fix plugin for Gnome 49 Compatibility

Unfortunately, the plugin used library versions that changed in Gnome 49, so
this migrates to use the new methods.

* [`Clutter.ClickAction()` -> `Clutter.ClickGesture()` and
  `Clutter.LongPressGesture()`](https://gjs.guide/extensions/upgrading/gnome-shell-49.html#clutter)
* [`OsdWindowManager.show()` -> `OsdWindowManager.showOne()`](https://gjs.guide/extensions/upgrading/gnome-shell-49.html#osdwindowmanager)
* bump version number in metadata.json

tmuncks commented:
Works for me! :)

JoelShepard commented:
It works but I can't access the "Nodes" submenu.
<img width="719" height="1324" alt="immagine" src="https://github.com/user-attachments/assets/d4beca10-0dd4-4e9d-97de-4bc46926902a" />

tmuncks commented:
Looks like perhaps you didn't apply the patch to the .ja file? Or perhaps you need to logout and log back in. 🙂

Works on Gnome 49 here.

JoelShepard commented:
It works. I used `make install` with `sudo`. Now is working perfectly

jee-r commented:
Tested on Gnome 49 (Fedora 43) no issues so far.

mattdale77 commented:
I'm just subscribing (can't work out how to do this without commenting on mobile)

keunes commented:
@mattdale77 In the app you have to tap on the top-right. On the mobile version of the website you have to go all the way to the bottom.

![Imagepipe_0.jpg](https://github.com/user-attachments/assets/f74dc627-8ed7-4463-beb5-3b4a2ed2afde) ![Imagepipe_0.jpg](https://github.com/user-attachments/assets/49434ed6-e5c5-48de-add6-e3804db6e293)

xuanruiqi commented:
Hello @joaophi , would you mind merging this and making a release? Thank you

kernelb00t commented:
@aoiwelle Can confirm this patch works on Gnome 49 / NixOS Unstable. 

@joaophi Is anything needed, or when would it be possible to merge this fix? 
We all know here open source can be time-consuming, if you need help maintaining this repo please consider telling us. I am sure a lot of people of trust would be excited to join you or take on the maintenance of the project. It's been a month and a half since this PR is opened, and it's sad that a useful extension like that goes dark. 

Thanks for this project, as usual open source cannot exist without your time !

aoiwelle commented:
> @aoiwelle Can confirm this patch works on Gnome 49 / NixOS Unstable.
> 
Yes, this works for Gnome 49. I just changed the versions supported to properly reflect that the patch only works for gnome 49. I also had noticed the OSD popup notifying that the IP was copied wasn't working before, so I just updated the change to restore that functionality as well. 

I've noticed some people have pulled this patch in with success. Is it possible for people to :rocket: this if it works for them,  or :-1:  and add a comment if they are having issues? This may help reduce the burden of review in a merging context.

spinogrizz commented:
Yep, works for me in Gnome 49, switching between two exit-nodes, everything as it was. 

Thanks for your effort!

kariudo commented:
This is a 🚀 as marked on the above comment, I did get hung up with (what might be nice to be a handled state) where if my user was not set as the operator they couldn't (after some debugging) query  `curl --unix-socket /var/run/tailscale/tailscaled.sock http://local-tailscaled.sock/localapi/v0/profiles` since it was rejected for permissions, so the plugin would seemingly load, but if you tried to change any setting or view profiles etc, it would crash and restart.

Would be nice if there was something to catch that permissions issue and warn they user to run:

`sudo tailscale set --operator=$USER` 

or similar, to ensure they have the required security set, (IIRC that might get warned to you if you open the tailscale GUI or something), but not this extension itself.

As soon as I did that, everything seemed to start working with this changeset. 👍🏻

kevinstsauveur commented:
> Would be nice if there was something to catch that permissions issue and warn they user to run:
> 
> `sudo tailscale set --operator=$USER`
> 
> or similar, to ensure they have the required security set, (IIRC that might get warned to you if you open the tailscale GUI or something), but not this extension itself.

@kariudo I agree with you, but I don't think this is something the majority of people are facing since it is clearly stated when you look at the description of the extension when you first install it on [https://extensions.gnome.org/extension/6139/tailscale-qs/](https://extensions.gnome.org/extension/6139/tailscale-qs/).
<img width="1132" height="275" alt="image" src="https://github.com/user-attachments/assets/957a7623-cf82-4176-bdf4-1b3172bc4112" />

kariudo commented:
> > Would be nice if there was something to catch that permissions issue and warn they user to run:
> > 
> > `sudo tailscale set --operator=$USER`
> > 
> > or similar, to ensure they have the required security set, (IIRC that might get warned to you if you open the tailscale GUI or something), but not this extension itself.
> 
> @kariudo I agree with you, but I don't think this is something the majority of people are facing since it is clearly stated when you look at the description of the extension when you first install it on [https://extensions.gnome.org/extension/6139/tailscale-qs/](https://extensions.gnome.org/extension/6139/tailscale-qs/).
> <img width="1132" height="275" alt="image" src="https://github.com/user-attachments/assets/957a7623-cf82-4176-bdf4-1b3172bc4112" />
> 

You're not wrong, I'm a believer in RTFM, but it's still not great behavior.

I have had my Op setting lost a couple times and had to remember to restore it. But I do run an immutable Linux distro, and constantly screw with my own images, so my environment has caused it to come up more times than it would most people. 

Certainly not a retirement, just a nice enhancement it could use when convenient. Not a complaint about the existing excellent extension.

alexjfinch commented:
Works perfectly for me on Fedora 43. Any chance it can be merged and pushed to gnome extensions?

jee-r commented:
It’s been almost two months since this pull request was opened.
If the repository owner doesn’t respond soon, I think we should consider forking the project under an organization with multiple maintainers and publishing it on GNOME Extensions.

What do you think?

alexjfinch commented:
I've checked and they're fairly active on other repos and their own private ones. Maybe they're not getting notifications from this?

kevinstsauveur commented:
For anyone coming here for a quick solution to manage Tailscale and who can't wait for this to be resolved, I would advise looking at Tailscale's new systray feature : [https://tailscale.com/kb/1597/linux-systray](https://tailscale.com/kb/1597/linux-systray).

BHSPitMonkey commented:
The other "quick" solution that works well is to uninstall the extension and then manually checkout or [download](https://github.com/aoiwelle/tailscale-gnome-qs/archive/refs/heads/fix-gnome-49.zip) the PR author's branch and manually `make build && make install` per the build instructions in the README; Just log out and log in afterward.

alexjfinch commented:
> It’s been almost two months since this pull request was opened. If the repository owner doesn’t respond soon, I think we should consider forking the project under an organization with multiple maintainers and publishing it on GNOME Extensions.
> 
> What do you think?

@aoiwelle Would you be interested in something like this? such a shame for a great extension to just disappear like this - ublue have forked it and are 50 odd commits ahead of this.

I sadly don't have any knowledge to be able to do this myself but would if I could!

aoiwelle commented:
> > It’s been almost two months since this pull request was opened. If the repository owner doesn’t respond soon, I think we should consider forking the project under an organization with multiple maintainers and publishing it on GNOME Extensions.
> > What do you think?
> 
> @aoiwelle Would you be interested in something like this? such a shame for a great extension to just disappear like this - ublue have forked it and are 50 odd commits ahead of this.
> 
> I sadly don't have any knowledge to be able to do this myself but would if I could!

Yeah, I'll create a fork under an org so more volunteers can help steward the plugin for usage. I'd been traveling so I haven't had a chance to submit the fork to Gnome Extensions.

xuanruiqi commented:
Is there a way to take over an extension on the GNOME extensions site?

k3dom commented:
For NixOS users who want to use this PR's version before it's merged, add the following to your configuration:

**flake.nix:**
```nix
{
  inputs = {
    # ... your other inputs
    tailscale-gnome-qs = {
      url = "github:aoiwelle/tailscale-gnome-qs/fix-gnome-49";
      flake = false;
    };
  };
}
```
In your system configuration (e.g., modules/system/nix.nix):
```nix
{
  nixpkgs.overlays = [
    (final: prev: {
      gnomeExtensions = prev.gnomeExtensions // {
        tailscale-qs = prev.gnomeExtensions.tailscale-qs.overrideAttrs (oldAttrs: {
          version = "49-unstable";
          src = "${inputs.tailscale-gnome-qs}/tailscale@joaophi.github.com";
        });
      };
    })
  ];
}
```

thewave1408 commented:
Any updates to the process of forking under a different extension?

zstoimchev reviewed (approved):
Tried on Fedora Linux 43 (Workstation Edition). Works as it should.

alexjfinch commented:
https://github.com/tailscale-qs/tailscale-gnome-qs

I've now created this under an organisation and added this PR so it now works on GNOME 49.

@aoiwelle do you want me to add you to the org? I don't want sole responsibility for maintaining this.

Will be submitting to gnome extensions this evening and will update the README.

jee-r commented:
Hi 
@alexjfinch It should probably be a fork of this repository, or at least mentioned in the README to credit @joaophi’s work. In my opinion.

alexjfinch commented:
> Hi @alexjfinch It should probably be a fork of this repository, or at least mentioned in the README to credit @joaophi’s work. In my opinion.

I've completely rewritten the README.md and Joaophi is absolutely credited for all of their hardwork - I'm just waiting for the gnome extension to be approved so I can update the links and will push change.

I've deforked it to allow it to continue cleanly.

aoiwelle commented:
I'm closing this in favor of [tailscale-qs/tailscale-gnome-qs](https://github.com/tailscale-qs/tailscale-gnome-qs) being the canonical fork. Thanks @alexjfinch!

aoiwelle commented:
> https://github.com/tailscale-qs/tailscale-gnome-qs
> 
> I've now created this under an organisation and added this PR so it now works on GNOME 49.
> 
> @aoiwelle do you want me to add you to the org? I don't want sole responsibility for maintaining this.
> 
> Will be submitting to gnome extensions this evening and will update the README.

@alexjfinch , feel free to add me to the organization. We should coordinate on how to share the load on the extension and how to roll out the extension as an Org and prevent any bus-number problems.
